python/ray/tune/trial_executor.py

```diff
@@ -1,14 +1,20 @@
 # coding: utf-8
+from abc import ABC, abstractmethod
 import logging
+from typing import Dict, List, Optional, TYPE_CHECKING, Union
 
+from ray.tune.resources import Resources
 from ray.tune.trial import Trial, Checkpoint
 from ray.tune.error import TuneError
 from ray.tune.cluster_info import is_ray_cluster
 
+if TYPE_CHECKING:
+    from ray.tune.trial_runner import TrialRunner
+
 logger = logging.getLogger(__name__)
 
 
-class TrialExecutor:
+class TrialExecutor(ABC):
```

Comment:
```suggestion
class TrialExecutor(metaclass=ABCMeta):
```

Reply:
Sounds good. Yeah I was going to wait for this one to be pushed first :)